### PR TITLE
W-14324937: fix tests and dependencies for java 17 (#474)

### PR DESCRIPTION
**Pull request by juanchib**

### CODE REVIEW

File: parser-service/src/test/java/org/mule/parser/service/ReferencesResolverTestCase.java [hunk 1]
```diff
@@ -18,12 +19,13 @@
...
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsEqual.equalTo;
-import static org.mockito.Matchers.any;
+import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.when;
 import static org.mule.apikit.model.api.ApiReference.create;
 
 public class ReferencesResolverTestCase {
 
+  @Ignore
```
Comment (JuanAller) [inline parser-service/src/test/java/org/mule/parser/service/ReferencesResolverTestCase.java:28]: is there a way to avoid @ignore?

**Review by santiagoGQ — DISMISSED** [review]

File: parser-service/src/test/java/org/mule/parser/service/ReferencesResolverTestCase.java (hunk 1 above)
Comment (juanchib) [reply parser-service/src/test/java/org/mule/parser/service/ReferencesResolverTestCase.java:28]: great catch Juan,, let's revert this until we can avoid it

**Review by iglosiggio — APPROVED** [review]

**Comment by iglosiggio** [comment]
LGTM

**Comment by JuanAller** [comment]
already included in a new support branch support/2.7.x

**Pull request closed without merging**